hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -293,7 +294,7 @@ boolean initInvocationTarget() {
         }
 
         targetMember = nodeEngine.getClusterService().getMember(invTarget);
-        if (!isJoinOperation(op) && targetMember == null) {
+        if (!(isJoinOperation(op) || isWanReplicationOperation(op)) && targetMember == null) {
```

Comment:
+ perhaps you might get a minor performance benefit by changing order of operands to
```java
targetMember == null && !(isJoinOperation(op) || isWanReplicationOperation(op))
```

as targetMember is not null in vast majority of cases - neither `isJoinOperation()` or `isWantReplicationOperation()` will be called thanks to short-circuiting.